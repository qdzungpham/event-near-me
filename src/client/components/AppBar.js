import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

const styles = {
  root: {
    flexGrow: 1
  },
  flex: {
    flexGrow: 1
  }
};

const SimpleAppBar = (props) => {
  const { classes, title } = props;
  return (
    <div className={classes.root}>
      <AppBar position="sticky">
        <Toolbar>
          <Typography className={classes.flex} variant="title" color="inherit">
            {title}
          </Typography>
          <Button href="https://github.com/qdzungpham/training-events-near-me" color="inherit">
            GITHUB
          </Button>
        </Toolbar>
      </AppBar>
    </div>
  );
};

SimpleAppBar.propTypes = {
  classes: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired
};

export default withStyles(styles)(SimpleAppBar);
