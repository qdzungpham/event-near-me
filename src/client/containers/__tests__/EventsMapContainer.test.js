import React from 'react';
import { shallow } from 'enzyme';
import AppBar from '../EventsMapContainer';

describe('EventMapContainer', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <AppBar
        events={[]}
        userLocation={{ lat: 12, lon: 12 }}
        onHoverMarker={jest.fn}
        overEventCard={jest.fn}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
