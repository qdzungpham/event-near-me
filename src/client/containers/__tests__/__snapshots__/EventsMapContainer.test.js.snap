// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventMapContainer renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withProps(withScriptjs(withGoogleMap(Component)))
    events={Array []}
    onHoverMarker={[Function]}
    overEventCard={[Function]}
    userLocation={
      Object {
        "lat": 12,
        "lon": 12,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "containerElement": <div
        style={
          Object {
            "height": "400px",
          }
        }
      />,
      "events": Array [],
      "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyDraTD9YAlK0aX50I_o-_L79g3DPirLYxA&v=3.exp&libraries=geometry,drawing,places",
      "loadingElement": <div
        style={
          Object {
            "height": "100%",
          }
        }
      />,
      "mapElement": <div
        style={
          Object {
            "height": "100%",
          }
        }
      />,
      "onHoverMarker": [Function],
      "overEventCard": [Function],
      "userLocation": Object {
        "lat": 12,
        "lon": 12,
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "containerElement": <div
          style={
            Object {
              "height": "400px",
            }
          }
        />,
        "events": Array [],
        "googleMapURL": "https://maps.googleapis.com/maps/api/js?key=AIzaSyDraTD9YAlK0aX50I_o-_L79g3DPirLYxA&v=3.exp&libraries=geometry,drawing,places",
        "loadingElement": <div
          style={
            Object {
              "height": "100%",
            }
          }
        />,
        "mapElement": <div
          style={
            Object {
              "height": "100%",
            }
          }
        />,
        "onHoverMarker": [Function],
        "overEventCard": [Function],
        "userLocation": Object {
          "lat": 12,
          "lon": 12,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
